packages/react-devtools-shared/src/devtools/views/Components/InspectedElementContext.js

```diff
@@ -198,7 +205,7 @@ function InspectedElementContextController({children}: Props) {
 
               fillInPath(inspectedElement, data.value, data.path, value);
 
-              resource.write(element, inspectedElement);
+              inspectedElementCache.set(element, inspectedElement);
```

Comment:
Does this happen if the item already exists in the cache? If yes this looks like a mutation and I'm not sure that's valid.

Reply:
Seems to work ok based on my testing.

Reply:
But might still be a bad idea :)

Reply:
I guess I don't know if it's any different from reading from a mutable store in general (if it's different, how?)

Reply:
Yeah, I need to think about this some more. I've got it listed as one of the TODOs on the PR. I just kind of swapped in place as much as possible with my initial commit. Good call to point it out though. 👍

Reply:
I guess the reason this "works" is that I'm duplicating the inspected element in local component state, and when these updates come in– I'm merging the update into both cache and state. (State is the thing I actually pass down through the context though.)

This was done originally to provide a mechanism to tell React that it needed to re-render after an inspected element update came from the backend. Happy to chat about alternate designs for this if you have ideas! :)